### PR TITLE
docs(proposal): produce CDEvents

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: docs/proposals/produce-cdevents.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+title: Neat-enhancement-idea
+authors:
+- @crenshaw-dev
+  sponsors:
+- TBD        # List all interested parties here.
```
Comment (afrittoli) [inline docs/proposals/produce-cdevents.md:6]: I'm interested, not an Argo contributor (yet) though.

File: docs/proposals/produce-cdevents.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  approvers:
+- TBD
+
+creation-date: 2023-05-24
+last-updated: 2023-05-24
+---
+
+# Produce CDEvents
+
+CDEvents is a standard for communicating about events in CD processes.
```
Comment (morey-tech) [inline docs/proposals/produce-cdevents.md:18]: Add a link for the lazy (and not in the know, i.e. me)
```suggestion
[CDEvents](https://cdevents.dev/) is a standard for communicating about events in CD processes.
```

**Comment by afrittoli** [comment]
@crenshaw-dev Thanks for this proposal! How can we move forward with this? Would it make sense to present the proposal at an ArgoCD working group?

**Comment by rjalander** [comment]
Like this proposal, we have made similar proposals for [Spinnaker](https://github.com/spinnaker/governance/blob/master/rfc/cdevents-spinnaker.md#produce-cdevents-1) and Flux(Spinnaker is up-streamed and Flux is in progress).
We are happy to participate in any contributions for Argo as well.

**Comment by crenshaw-dev** [comment]
Gonna close this for now, the CDEvents folks I think plan to build the initial PoC using Argo Notifications with JSON payloads.

**Pull request closed without merging**